src/common/object/jsc/object.hpp

```diff
@@ -0,0 +1,230 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+// This allow us to run the JSC tests on our Mac's locally.
+#if __APPLE__
+#include <JavaScriptCore/JavaScriptCore.h>
+#endif
+
+#include <iostream>
+#include <vector>
+
+#include "common/object/error_handling.hpp"
+#include "common/object/interfaces.hpp"
+#include "realm/object-store/shared_realm.hpp"
+
+namespace realm {
+namespace common {
+
+struct PrivateStore {
+    void *accessor_data = nullptr;
+    ObjectObserver *observer = nullptr;
+    IOCollection *collection = nullptr;
+    std::function<void()> finalizer = nullptr;
+    std::unordered_map<std::string, bool> keys;
+};
+
+template <typename GetterSetter>
+class JavascriptObject {
+   private:
+    JSClassDefinition _class;
+    JSContextRef context;
+    JSObjectRef object{nullptr};
+    std::vector<JSStaticFunction> methods;
+    std::vector<std::string> accessors;
+    PrivateStore *private_object;
+
+    static std::string to_string(JSContextRef context, JSStringRef value) {
```

Comment:
Because that one is not reusable, unless you pull the full global variable in the form of template type. 

In order to make this module self contain and testable I/we need to avoid pulling global variables in disguise like the one necessary to reuse the function you mention.